include/alpaka/exec/ExecCpuOmp2Blocks.hpp

```diff
@@ -148,56 +148,72 @@ namespace alpaka
                     throw std::runtime_error("Only one thread per block allowed in the OpenMP 2.0 block accelerator!");
                 }
 
-                // Force the environment to use the given number of threads.
-                int const ompIsDynamic(::omp_get_dynamic());
-                ::omp_set_dynamic(0);
-
-                // Execute the blocks in parallel.
-                // NOTE: Setting num_threads(number_of_cores) instead of the default thread number does not improve performance.
-                #pragma omp parallel
+                if (omp_in_parallel() != 0)
+                {
+                    parallelFn(
+                        boundKernelFnObj,
+                        blockSharedMemDynSizeBytes,
+                        numBlocksInGrid,
+                        gridBlockExtent);
+                }
+                else
                 {
+                    #pragma omp parallel
+                    parallelFn(
+                        boundKernelFnObj,
+                        blockSharedMemDynSizeBytes,
+                        numBlocksInGrid,
+                        gridBlockExtent);
+                }
+            }
+
+            template<
+                typename FnObj>
+            ALPAKA_FN_HOST auto parallelFn(
+                const FnObj& boundKernelFnObj,
```

Comment:
Shouldn't be the `const on the right side of the type as for the fourth parameter?

Reply:
Yes, I will fix this. Simply copied it from the proposal.